components/esp_system/panic.c

```diff
@@ -81,14 +81,25 @@ static void panic_print_char_uart(const char c)
 }
 #endif // CONFIG_ESP_CONSOLE_UART
 
+#if CONFIG_ESP_CONSOLE_SECONDARY_UART_DEFAULT
```

Comment:
Right. I forgot I added `&& !ESP_CONSOLE_UART_CUSTOM`. 

I added that because so you can't accidentally configure a `CUSTOM_UART` and a `SECONDARY_UART_DEFAULT` to use the same port. 

So I added `&& !ESP_CONSOLE_UART_CUSTOM` to simplify implementation.

Perhaps I should use fancier `KConfig` settings to solve this? Something like `&& !(ESP_CONSOLE_UART_CUSTOM && UART_NUM == 0)`

Reply:
updated.

Reply:
They can exist at the same time.

You could have this configuration:
- ESP_CONSOLE_UART_CUSTOM & UART_NUM_1
- ESP_CONSOLE_SECONDARY_UART_DEFAULT

Reply:
Understood. 

Having them treated separately (as my code currently does), is IMO the simpler approach.

I'd need to think more about how they could be combined, given that the primary implementation has configurable GPIO pins, UART Num, etc. I would need to make sure those values are set correctly.

Reply:
Actually, there is a super great use-case for 2 console - independent log processing & storage on a separate microcontroller.

This is actually something I considered implementing myself.

Logs are super valuable, and in remote locations you need a way to access logs, especially when your main microcontroller is misbehaving.

Being able to send logs over UART to a 2nd microcontroller is super useful.

Reply:
Ya that seems smarter. I suppose that should work easily for the TX-pin.

I'll look more into combining them.